OneSignalSDK/onesignal/src/main/java/com/onesignal/LocationController.java

```diff
@@ -289,106 +269,38 @@ static void startGetLocation() {
       }
    }
 
-   private static void initGoogleLocation() {
-      // Prevents overlapping requests
-      if (fallbackFailThread != null)
-         return;
-
-      synchronized (syncLock) {
-         startFallBackThread();
-
-         if (googleApiClient == null || lastLocation == null) {
-            GoogleApiClientListener googleApiClientListener = new GoogleApiClientListener();
-            GoogleApiClient googleApiClient = new GoogleApiClient.Builder(classContext)
-                    .addApi(LocationServices.API)
-                    .addConnectionCallbacks(googleApiClientListener)
-                    .addOnConnectionFailedListener(googleApiClientListener)
-                    .setHandler(locationHandlerThread.mHandler)
-                    .build();
-
-            LocationController.googleApiClient = new GoogleApiClientCompatProxy(googleApiClient);
-            LocationController.googleApiClient.connect();
-         } else if (lastLocation != null)
-            fireCompleteForLocation(lastLocation);
-      }
-   }
-
-   private static void initHuaweiLocation() {
+   static void onFocusChange() {
       synchronized (syncLock) {
-         if (huaweiFusedLocationClient == null) {
-            try {
-               huaweiFusedLocationClient = com.huawei.hms.location.LocationServices.getFusedLocationProviderClient(classContext);
-            } catch (Exception e) {
-               OneSignal.Log(OneSignal.LOG_LEVEL.ERROR, "Huawei LocationServices getFusedLocationProviderClient failed! " + e);
-               fireFailedComplete();
-               return;
-            }
+         // Google location not initialized or connected yet
```

Comment:
We should check for location permission here before attempting to access `GMSLocationController` or `HMSLocationController`.

Otherwise `NoClassDefFoundError` will print in the logcat if they are don't have the location library in their app.

Reply:
good catch! Added dependency check